### DISCUSSION TITLE
how can I intercept the SK outbound https traffic for debugging and troubleshooting? such as using mitmweb/mitmproxy

### DISCUSSION THREAD

**Discussion by AwesomeYuer** [microsoft/semantic-kernel#1513] [Q&A] [answered]
Would confirm SK use http/2.0 or other special setting for outbound accessing Azure OpenAI endpoint?

Use Azure OpenAI

![image](https://github.com/microsoft/semantic-kernel/assets/1026479/f8a703b2-c059-40bd-a276-47c71ba9ff4a)

When I running all the cells in `00-getting-started.ipynb` on Windows 11 OS

I set the HTTP Proxy  to  `mitmweb/mitmproxy` as below code in one cell:

`
using System.Net.Http;

Environment.SetEnvironmentVariable("http_proxy", "http://localhost:8080");
Environment.SetEnvironmentVariable("https_proxy", "http://localhost:8080");
`

and set proxy on the Windows 11 proxy settings:

![image](https://github.com/microsoft/semantic-kernel/assets/1026479/8f953901-600a-4f9a-8f20-a621aa54f959)

then I run cells one by one, after run the cell of  `step 4 Load and Run a Skill`, I got the success  result as below:
`
A time traveler went back to the dinosaur age and was amazed by the size of the creatures. He asked one of the dinosaurs, "How do you manage to get around with such short legs?"

The dinosaur replied, "It's easy, I just take my time!"
`
On the UI of mitmproxy that's mitmweb, can't find any captured entries.

But `mitmproxy/mitmweb`  support `https/http http/2.0`  traffic intercepting and decrypting, It can  intercept and decrypt  the traffic from browser(chrome or edge).

![image](https://github.com/microsoft/semantic-kernel/assets/1026479/adda9947-8a9d-4728-a781-d6f2a39fb89b)

BTW： fiddler classic don't support  `http/2.0`,  so use `mitmproxy/mitmweb`

https://github.com/mitmproxy/mitmproxy

**Comment by craigomatic** [comment, marked as answer]
Perhaps the Environment.SetEnvironmentVariable calls are not propagating?

Another approach you could try:

```csharp
//give the HttpClient an instance of your proxy via the HttpClientHandler
var httpClient = new HttpClient(new HttpClientHandler { Proxy = new MyProxy() });

var sk = Kernel.Builder.WithAzureChatCompletionService(
"<deployment name>", 
"<endpoint>", 
"<key>", 
true, 
null, 
false, 
httpClient); //the HttpClient you pass in here is used by the underlying completion methods

//change the GetProxy URI below if it differs from what you need
class MyProxy : IWebProxy
{
    public ICredentials? Credentials { get; set; }

    public Uri? GetProxy(Uri destination)
    {
        return new Uri("http://localhost:8080");
    }

    public bool IsBypassed(Uri host)
    {
        return false;
    }
}
```

**Reply by AwesomeYuer** [reply]
@craigomatic  thanks
